src/crafting.cpp

```diff
@@ -1831,33 +1689,7 @@ void player::complete_disassemble( int item_pos, const tripoint &loc,
 
 const recipe *recipe_by_name( const std::string &name )
```

Comment:
Indeed one of the goals is to get rid of the pointer shenanigans. Would you prefer I attempted for this in a single PR?